### PR TITLE
Bug 1400949 - Password requirements not stated

### PR DESCRIPTION
**Pull request by dylanwh**

### CODE REVIEW

File: Bugzilla/Config/Auth.pm [hunk 1]
```diff
@@ -17,6 +17,14 @@ use Types::Common::Numeric qw(PositiveInt);
 
 our $sortkey = 300;
 
+use constant PASSWDQC_DESC => <<'DESC';
+Password containing only one class of characters is not allowed.<br>
+Password containing two classes of characters must be 24 characters or longer<br>
+Password that is a passphrase must be 11 characters or longer, and at least 3 words.<br>
+Password containing three classes of characters must be 8 characters or longer.<br>
+Password containing four classes of characters must be 7 characters or longer.
```
Comment (globau) [inline Bugzilla/Config/Auth.pm]: this doesn't describe what a "class of character" is.

i'm concerned there's a lot of dense information here; although a lot of it should be not required with `undef,undef,12,12,12`.

perhaps it would be better to have a simpler "dumbed down" explanation here, which provides enough of a hint for most users, with a link to a wiki page that contains the full technical breakdown for each character class.
eg.

> Your password must be at least <b>12 characters long</b>, and contain:
> * a mixture of uppercase, lowercase, and numbers
> OR
> * a mixture or symbols, numbers, and letters
> OR
> * a phrase with at least four words
> 
> <a href="#">View our full password complexity rules.</a>

File: template/en/default/account/email/confirm-new.html.tmpl [hunk 2]
```diff
@@ -26,13 +26,14 @@
...
 
-[% password_complexity = Param('password_complexity') %]
-
 <p>
   To create your account, you must enter a password in the form below.
   Your email address and Real Name (if provided) will be shown with
   changes you make.
 </p>
+<p>
+  [% Param('passwdqc_desc') FILTER none %]
```
Comment (globau) [inline template/en/default/account/email/confirm-new.html.tmpl]: this should probably be `html_light` instead of `none`.

File: template/en/default/account/password/set-forgotten-password.html.tmpl [hunk 3]
```diff
@@ -25,7 +25,10 @@
  %]
 
 <p>
-  To change your password, enter a new password twice:
+  To change your password, enter a new password twice.
+</p>
+<p>
+  [% Param('passwdqc_desc') FILTER none %]
```
Comment (globau) [inline template/en/default/account/password/set-forgotten-password.html.tmpl]: `html_light` here too.

File: template/en/default/account/prefs/account.html.tmpl [hunk 4]
```diff
@@ -71,6 +71,12 @@
         </tr>
         [% IF user.authorizer.can_change_password %]
           <tr>
+          <td>&nbsp;</td>
+          <td>
+            [% Param('passwdqc_desc') FILTER none %]
```
Comment (globau) [inline template/en/default/account/prefs/account.html.tmpl]: `html_light` here too.

File: template/en/default/account/reset-password.html.tmpl [hunk 5]
```diff
@@ -75,6 +75,10 @@ $(function() {
   [% user.password_change_reason || "You are required to update your password." FILTER html %]
 </p>
 
+<p>
+  [% Param('passwdqc_desc') FILTER none %]
```
Comment (globau) [inline template/en/default/account/reset-password.html.tmpl]: `html_light` here too.

File: template/en/default/admin/params/auth.html.tmpl [hunk 6]
```diff
@@ -87,6 +87,9 @@
   The size of randomly-generated passphrases in bits (24 to 85).
 [% END %]
 
+[% desc_passwdqc_desc = BLOCK %]
+  This should be a short paragraph describing the password requirements in plain English.
```
Comment (globau) [inline template/en/default/admin/params/auth.html.tmpl]: add "Limited HTML is permitted."

File: Bugzilla/Config/Auth.pm (hunk 1 above)
Comment (dylanwh) [reply Bugzilla/Config/Auth.pm]: I changed the default message to something terse. The revised message will be part of configuration.
